docs/development/pigs/pig-014.rst

```diff
@@ -0,0 +1,158 @@
+differential and integral flux of spectral models. This is cumbersome since
+``uncertainties`` doesn't support ``astropy.units.Quantity`` (which we currently
+use in spectral model evaluation), and doesn't work at all for some spectral
+models, since ``uncertainties`` uses autodiff and needs explicit support for any
+Numpy, Scipy, Astropy, ... function involved.
+
+We propose to replace this with a custom uncertainty propagation implementation
+using finite differences. This will be less precise and slower than
+``uncertainties``, but it will work for any derived quantity and code that we
+call.
+
+We also propose to add support for a second method to propagate uncertainty from
+fit parameters to derived quantities, based on parameter samples. This is the
+standard technique in MCMC Bayesian analyses, but it can be used as well for
+normal likelihood analyses.
+
+Introduction
+============
+
+In Gammapy, ``gammapy.modeling`` and specifically the ``gammapy.modeling.Fit``
+class support likelihood fitting to determine best-fit parameters, and to obtain
+a covariance matrix that represents parameter uncertainties and correlations.
+Parameter standard errors are obtained as the square root of the elements on the
+diagonal of the covariance matrix (see e.g. `The interpretation of errors`_).
+Asymmetric errors and confidence intervals can be obtained via likelihood
+profile shape analysis, using methods on the ``gammapy.modeling.Fit` class,
+partially re-implementing, partially calling the methods available in Minuit or
+Sherpa (see e.g. `Fitting and Estimating Parameter Confidence Limits with
+Sherpa`_).
+
+However, often one is interested not directly in a model parameter, but instead
+in a derived quantity that depends on multiple model parameters. Typical
+examples are that users want to compute errors or confidence intervals on
+quantities like differential or integral flux at or above certain energies, or
+location or extension of sources, from spectral and spatial model fits.
+
+Here we will discuss two standard techniques to compute such uncertaintes:
+
+1. Differentials (see `Wikipedia - Propagation of uncertainty`_ or `uncertainties`_)
+2. Monte carlo (MC) samples (see `mcerp`_ or `astropy.uncertainty`_)
+
+If you're not familiar with the techniques, here's a few references:
+
+- `Error estimation in astronomy: A guide`_
+- `Frequentism and Bayesianism: A Python-driven Primer`_
+
+In Gammapy we currently use the `uncertainties`_ package to propagate errors to
+differential fluxes ``spectral_model.evaluate_error`` and integral fluxes
+``spectral_model.integral_error``. It is using autodiff to compute differentials
+of derived quantities wrt. model parameters, which is accurate and fast where it
+works. However, ``uncertainties`` doesn't support ``astropy.units.Quantity``,
+making the current spectral model flux evaluation very convoluted, and it
+doesn't work at all for some spectral models, which has been a frequent
```

Comment:
you could explicitly state that is fails for non analytical models, which is the case for most of physical models (e.g. EBL absorption or naima models)